spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -419,6 +419,9 @@ else if (listener instanceof MessageListener) {
 			}
 			this.monitorTask = this.taskScheduler.scheduleAtFixedRate(() -> checkConsumer(),
 					this.containerProperties.getMonitorInterval() * 1000);
+			Assert.state(this.transactionTemplate == null
```

Comment:
I would say that it is better to move this assert before that `monitorTask` scheduling